### PR TITLE
Stage1

### PR DESCRIPTION
**Pull request by roeebenezra**

### CODE REVIEW

File: main_stage1.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    presentation_path = args.presentation_path
+
+    print(f"Processing the pptx: {presentation_path}")
+
+    # Parse presentation
+    presentation_parser = PresentationParser(presentation_path)
+    slides = presentation_parser.process_presentation()
+
+    # Generate explanations
+    api_key = os.environ.get("OPENAI_API_KEY")
```
Comment (excellenteam-scaleup) [inline main_stage1.py:40]: This is not the place to do that.
I didn't have a key and I needed to scan your code to find how you want me to enter the key.
Instead you can have it as a constant in the top of the file, or even in a config file

File: main_stage1.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    """
+    presentation_name = os.path.splitext(os.path.basename(presentation_path))[0]
+    output_file = f"{presentation_name}.json"
+    output_data = {"explanations": explanations}
+
+    with open(output_file, "w") as f:
+        json.dump(output_data, f, indent=4)
+
+
+def main():
```
Comment (excellenteam-scaleup) [inline main_stage1.py:26]: The best way is to define main as async def
https://docs.python.org/3/library/asyncio-task.html

File: openAi.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+        """
+        messages = self.messages + [{"role": "user", "content": slide}]
+        data = {
+            "engine": "text-davinci-003",
+            "prompt_messages": messages,
+            "max_tokens": 50,
+            "n": 1,
+            "stop": None,
+            "temperature": 0.7,
+            "timeout": 60,
```
Comment (excellenteam-scaleup) [inline openAi.py:61]: This is also a good example for config file variables

File: openAi.py [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+    """
+
+    def __init__(self, api_key):
+        """
+        Constructor for OpenAIAPI class.
+        :param api_key: OpenAI API key
+        """
+        self.api_key = api_key
+        self.messages = [
+            {"role": "system", "content": "Please summarize the slides and provide additional information."}
```
Comment (excellenteam-scaleup) [inline openAi.py:20]: You give very little context to chatGPT.
Writing prompts is a new art we all need to learn.

File: test.pptx
Comment (excellenteam-scaleup) [inline test.pptx:1]: Your test is very basic. It shows that text can be extracted from the pptx, but there is no content so you can't see if GPT even understands you.
In addition you should test edge-cases like blank slides and text-boxes

File: main_stage1.py
Comment (excellenteam-scaleup) [inline main_stage1.py:1]: main func should simply be called main

File: openAi.py [hunk 5]
```diff
@@ -0,0 +1,66 @@
+import asyncio
+
```
Comment (excellenteam-scaleup) [inline openAi.py:2]: redundant empty line

**Review by excellenteam-scaleup — APPROVED** [review]
Good job.
Seem you managed to explore new packages and use them.
Code is divided into separate modules and functions, well documented, and follow PEP8.
I hope your system will get more real and functional in the next stages.
I noticed you did everything in few commits. when we work with git it's better to commit every time you finished a logical part, for example every function. Also each module should get it's own branch.

Keep up the good work!
